lib/components/auth_state.dart

```diff
@@ -25,8 +28,24 @@ class AuthState<T extends StatefulWidget> extends SupabaseAuthState<T> {
   }
 
   @override
-  void onPasswordRecovery(supabase.Session session) {
+  void onErrorAuthenticating(String message) {
+    alertModal.show(context, title: 'Sign in failed', message: message);
+  }
+
+  @override
+  void onPasswordRecovery(Session session) {
     // TODO: show password change screen
     throw UnimplementedError();
   }
+
+  @override
+  void onResumed() {
+    if (signingInWithOAuthProvider && !deeplinkReceived) {
+      /// User pressed the back button of the browswer to come back to the app
+      onErrorAuthenticating('Could not authenticate');
```

Comment:
This is here to catch when a user pressed the back button on their browser, but not sure if auth_state.dart is the right place for it.